README.md
# LoveBoard


the real love social network

### mision

a dating app that doesnt make you sad or feel axiety
### vision


## objetives
focus on user interface customizations
focus on mobile first
focus on be a free app without paid addons

## TODO
    - demo in english
      - core features
        - your board edition
          - add images/videos/text/stickers
          - modify colors
        - see other boards
        - save user profile on local storage/ save to json
      - backend stuff
        - accounts
        https://ory.sh/
        - save media - file storage
        https://min.io/
      - dev community
        - document all the things
        - make sure to have all the code available for everyone, remenber is open source
      - make the proyect sustainable
        - add adds
        - buy domain
        - deploy yo production cloud 
    - 
    - multi languaje
    - user experience
    - user interface
    - authentication with google/email/facebook/microsoft
    - deploy to prod

CHANGELOG.md

(backlog)

(current)
V 0.0.1
# grid
    card
        when you click in other div disable editing mode

(coming soon)

src/App.js
import { useState } from "react";

import Grid from "./components/Grid";
import Card from "./components/Card";
import Widgets, { WIDGETS } from "./components/Widgets";
import BottomNavbar from "./components/mmm/BottomNavbar";


export default function App() {
  const [widgets, setWidgets] = useState([]);

  return (
    <>
    <Grid setWidgets={setWidgets}>
      {(actions) =>
        widgets.map((widget) => {
          const { component: Widget, label } = WIDGETS[widget.type];

          return (
            Widget && (
              <Card
                key={widget.id}
                actions={actions}
                title={label}
                {...widget}
              >
                <Widget />
              </Card>
            )
          );
        })
      }
    </Grid>
      <BottomNavbar>
        <Widgets />
      </BottomNavbar>
    </>
  );
}


src/components/Card.js
import { useEffect, useRef, useState } from "react";

import onLongpress from "./../scripts/cardsActions";
import "./Card.css";
import EditToolbar from "./subcomponents/editToolbar/EditToolbar";
/*
export const onDoubleTap = function(el, handler) {
  const MAX_TIME_TO_SECOND_TAP = 300 // espera máximo 300ms al siguiente click/tap
  let lastTapTimestamp = 0;
  el.addEventListener('touchstart', function() {
    lastTapTimestamp = new Date().getTime();
  });
  el.addEventListener('touchend', function(event) {
      var currentTime = new Date().getTime();
      var tapEllapsedTime = currentTime - lastTapTimestamp;

      // si se ha hecho el segundo tap/click entre los 300ms
      if (tapEllapsedTime < MAX_TIME_TO_SECOND_TAP && tapEllapsedTime > 0) {
          event.preventDefault();
          lastTapTimestamp = 0; // reinicia el counter
          handler(event); // ejecuta el handler
      }
  });
};
*/

/*
taphold | long press

https://interactjs.io/
https://jsfiddle.net/kelunik/pkjze6e6/42/
https://stackoverflow.com/questions/2625210/long-press-in-javascript
https://www.w3schools.com/jquerymobile/tryit.asp?filename=tryjqmob_events_taphold
https://stackoverflow.com/questions/41329964/click-and-hold-onto-a-selected-div-to-hide
https://www.npmjs.com/package/long-press-event


https://codepen.io/allurewebsolutions/pen/YqJyYY
https://codepen.io/borntofrappe/pen/jOEKERG


*/

export default function Card({
  id,
  title: initialTitle,
  w,
  h,
  x,
  y,
  actions,
  children,
}) {
  const cardRef = useRef(null);
  const [toggle, setToggle] = useState(true);
  const [title, setTitle] = useState(
    initialTitle || "Double click to change title"
  );
  const [visible, setVisible] = useState(true);
  const [editable, setEditable] = useState(false);

  useEffect(() => {
    actions.handleRemove(cardRef.current, false);
    actions.handleAdd(cardRef.current);

    onLongpress(cardRef.current,handleEditCard);

    //https://www.codegrepper.com/code-examples/javascript/get+cursor+position+javascript
    var pointerX = -1;
    var pointerY = -1;
    document.onmousemove = function (event) {
      pointerX = event.pageX;
      pointerY = event.pageY;
    };
  }, []);

  const handleToggle = (flag) => {
    setToggle(flag);
    actions.handleEnableMove(flag);
  };
  const handleTapHold = (flag) => {
    //setToggle(flag);
    //actions.handleEnableMove(flag);
    console.log("handleTapHold");
  };
  /*
  var changeName = () => {
    console.log("changeName");
  };
  function changeName(evt) {
    console.log("changeName", evt);
  }
*/
  const changeName = (evt) => {
    console.log("changeName", evt);
  };

  const backCard = (evt) => {
    console.log("backCard", evt);
    setEditable(false);
    //cardRef.current.classList.remove("longpress");
    cardRef.current.classList.remove('longpress');
    cardRef.current.classList.remove("itemselected");
    //cardRef.current.class.remove('longpress');
    //cardRef.current.className.remove('longpress');
    //console.log("cardRef.current", cardRef.current);
    //console.log("cardRef", cardRef);


    //document.getElementById(id).classlist.remove('longpress');
    //cardRef.classList.remove('longpress');
    //cardRef.current.classList.add("longpress2");
  };
  const editCard = (evt) => {
    console.log("editCard", evt);
  };
  const handleEditCard = (flag) => {
    setEditable(true);
    console.log("handleEditCard", flag);
  };


  const delCard = (evt) => {
    console.log("delCard", evt);
    //gridRef.current.removeWidget(el, false);
    actions.handleRemove(cardRef.current, false);
    //this.props.unmountMe();
    //ReactDOM.unmountComponentAtNode(ref.current);
    setVisible(false);
  };
  return visible ? (
    <div
      ref={cardRef}
      id={`${id}`} // convert to string
      className="grid-stack-item"
      gs-w={w}
      gs-h={h}
      gs-x={x}
      gs-y={y}
    >
      {/**
       * Toolbar
       */}
      <div className="grid-stack-item-content">
        {editable ? (<EditToolbar className="" delCard={delCard} editCard={editCard} backCard={backCard} ></EditToolbar>):(<div/>)}
        {children}
        {/*
        <header>
          {toggle ? (
            <h2
              title="Double click to change title"
              onDoubleClick={() => handleToggle(false)}
            >
              {title}
            </h2>
          ) : (
            <input
              autoFocus
              type="text"
              value={title}
              onChange={(event) => setTitle(event.target.value)}
              onKeyDown={(event) => {
                if (event.key === "Enter" || event.key === "Escape") {
                  handleToggle(true);
                  event.preventDefault();
                  event.stopPropagation();
                }
              }}
              onBlur={() => handleToggle(true)}
            />
          )}

          <button
            title="Delete widget"
            onClick={() => {
              actions.handleRemove(cardRef.current);
            }}
          >
            &#x2715;
          </button>
        </header>
         */}
      </div>
    </div>
  ) : (
    <div />
  );
}


src/components/Grid.js
import { useRef, useEffect } from "react";
//import { GridStack } from "gridstack";

import GridStack from "gridstack/dist/gridstack-jq";

//import "gridstack/dist/gridstack-jq";
import "gridstack/dist/gridstack.css";

import "./Grid.styles.scss";
import "./Grid.css";

const Item = ({ id }) => <div>I am item: {id}</div>;

let id = 1;
export default function Grid({ children, setWidgets }) {
  const refs = useRef({});
  const gridRef = useRef();

  useEffect(() => {
    gridRef.current = GridStack.init({
      column: 6, // will auto switch on smaller screens
      row:10,
      disableOneColumnMode: true,
      float: true,
      acceptWidgets: true,
      //draggable:{handle:'.drag-handle'},
      removable: true,
      staticGrid: false,
      //cellHeight: 'initial', // start square but will set to % of window width later

      //cellHeightUnit: "px",
      //cellHeight: 70, // make sure we have a decent height and not width/12 for 1 column
      alwaysShowResizeHandle: /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(
        navigator.userAgent
      ),

      cellHeight: 'auto',
      //cellHeight: 70,
      minRow: 4, // don't collapse when empty
      margin: 4,
      resizable: {
        handles: "all",
      },
      dragIn: ".droppable", // class that can be dragged from outside
      dragInOptions: {
        revert: "invalid",
        scroll: false,
        appendTo: "body",
        helper: "clone",
      },
      //column: 6, // will auto switch on smaller screens
      //row: 10,
      //cellHeight: "auto", // start square but will set to % of window width later
      //cellWidth:"initial",
    });

    const grid = gridRef.current;
    grid.load([
      { x: 0, y: 0, content: "1" },
      { x: 0, y: 1, h: 2, content: "2" },
      { x: 0, y: 3, content: "3" },
    ]);

    if (grid) {
      grid.on("dropped", function (event, previousWidget, newWidget) {
        const { el, w, h, x, y } = newWidget;
        grid.removeWidget(el);
        setWidgets((items) => [
          ...items,
          {
            id: id++,
            type: el.dataset.type,
            w,
            h,
            x,
            y,
          },
        ]);
      });
    }
  }, []);

  const handleAdd = (el) => {
    if (el && gridRef.current) {
      gridRef.current.makeWidget(el);
    }
  };
  const handleRemove = (el, actualRemove = true) => {
    if (el && gridRef.current) {
      gridRef.current.removeWidget(el, false);
      actualRemove &&
        setWidgets((items) => items.filter((item) => `${item.id}` !== el.id));
    }
  };
  const handleEnableMove = (flag = true) => {
    if (gridRef.current) {
      gridRef.current.enableMove(flag);
    }
  };

  return (
    <section>

      <div id="stacker" className="grid-stack">
        {children({
          handleAdd,
          handleRemove,
          handleEnableMove,
        })}
      </div>
    </section>
  );
}


src/components/Grid.1.js
import { useRef, useEffect } from "react";
//import { GridStack } from "gridstack";

import GridStack from "gridstack/dist/gridstack-jq";

//import "gridstack/dist/gridstack-jq";
//import "gridstack/dist/gridstack.css";

// import "./styles.scss";
import "./demo.css";

const Item = ({ id }) => <div>I am item: {id}</div>;

let id = 1;
export default function Grid({ children, setWidgets }) {
  const refs = useRef({});
  const gridRef = useRef();
  /*
  
      column: 6, // will auto switch on smaller screens
      row: GridRows,
      disableOneColumnMode: true,
      float: true,
      acceptWidgets: true,
      resizable: { handles: "e, se, s, sw, w" },
      //draggable:{handle:'.drag-handle'},
      removable: true,
      staticGrid: false,
      cellHeight: UltimateCellHeight, // start square but will set to % of window width later
      cellWidth:"initial",
      //cellHeightUnit: "px",
      //cellHeight: 70, // make sure we have a decent height and not width/12 for 1 column
      alwaysShowResizeHandle: /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(
        navigator.userAgent
      )
  
  */
  useEffect(() => {
    gridRef.current = GridStack.init({
      column: 6, // will auto switch on smaller screens
      row:10,
      disableOneColumnMode: true,
      float: true,
      acceptWidgets: true,
      //draggable:{handle:'.drag-handle'},
      removable: true,
      staticGrid: false,
      //cellHeight: 'initial', // start square but will set to % of window width later

      //cellHeightUnit: "px",
      //cellHeight: 70, // make sure we have a decent height and not width/12 for 1 column
      alwaysShowResizeHandle: /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(
        navigator.userAgent
      ),

      //cellHeight: 'auto',
      cellHeight: 70,
      minRow: 4, // don't collapse when empty
      margin: 4,
      resizable: {
        handles: "all",
      },
      dragIn: ".droppable", // class that can be dragged from outside
      dragInOptions: {
        revert: "invalid",
        scroll: false,
        appendTo: "body",
        helper: "clone",
      },
      //column: 6, // will auto switch on smaller screens
      //row: 10,
      //cellHeight: "auto", // start square but will set to % of window width later
      //cellWidth:"initial",

    }, '#stacker');

    const grid = gridRef.current;
    grid.load([
      { x: 0, y: 0, content: "1" },
      { x: 0, y: 1, h: 2, content: "2" },
      { x: 0, y: 3, content: "3" },
    ]);

    if (grid) {
      grid.on("dropped", function (event, previousWidget, newWidget) {
        const { el, w, h, x, y } = newWidget;
        grid.removeWidget(el);
        setWidgets((items) => [
          ...items,
          {
            id: id++,
            type: el.dataset.type,
            w,
            h,
            x,
            y,
          },
        ]);
      });
    }
  }, []);

  const handleAdd = (el) => {
    if (el && gridRef.current) {
      gridRef.current.makeWidget(el);
    }
  };
  const handleRemove = (el, actualRemove = true) => {
    if (el && gridRef.current) {
      gridRef.current.removeWidget(el, false);
      actualRemove &&
        setWidgets((items) => items.filter((item) => `${item.id}` !== el.id));
    }
  };
  const handleEnableMove = (flag = true) => {
    if (gridRef.current) {
      gridRef.current.enableMove(flag);
    }
  };

  return (

      <div id="stacker" className="grid-stack nomelacontainer">
        {children({
          handleAdd,
          handleRemove,
          handleEnableMove,
        })}
      </div>
  );
}


src/components/Widgets.js
export const WIDGETS = {
  grid: {
    component: () => <div style={{ textAlign: "center" }}>GRID</div>,
    label: "Grid",
  },
  chart: {
    component: () => <div className="grid-stack-item-content">CHART</div>,
    label: "Chart",
  },
  photo: {
    component: () => (
      <div style={{ textAlign: "center" }}>
        <img src="https://placeimg.com/100/100/animals" alt="michi" />
      </div>
    ),
    label: "Photo",
  },
  video: {
    component: () => (
      <div style={{ textAlign: "center" }}>
        <div className="grid-stack-item-content" style={{ padding: "0px" }}>
          <iframe
            width="100%"
            height="100%"
            src="https://www.youtube-nocookie.com/embed/BRGwCgfS4Qk?autoplay=0"
            title="YouTube video player"
          ></iframe>
        </div>
      </div>
    ),
    label: "Video",
  }
};
{/*
frameborder="0" 
            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
            allowfullscreen

*/}
import "./Widgets.css";

export default function Widgets() {
  return (
    <aside className="grid-container">
      {Object.entries(WIDGETS).map(([key, value]) => (
        <div
          key={key}
          data-type={key}
          className="droppable grid-stack-item photo-grid-stack-item grid-item"
          gs-h="2"
          gs-w="2"
        >
          <div className="grid-stack-item-content">
            <div>
              {/*
              <span>Let me in, Im a photo</span>      
              */}
              {value.label}
            </div>
          </div>
        </div>
      ))}
      {/*
      <div
        className="droppable grid-stack-item photo-grid-stack-item"
        key="photo3"
        data-type="photo3"
        gs-h="2"
        gs-w="2"
      >
        <div className="grid-stack-item-content">
          <div>
            <span>Let me in, Im a photo</span>
          </div>
          <div>
            <span>Let me in, Im a photo</span>
          </div>
        </div>
      </div>
      */}
    </aside>
  );
}


src/components/mmm/BottomNavbar.js
import anime from "animejs/lib/anime.es.js";

import "./BottomNavbar.css";

const buttons = document.querySelectorAll(".bottom-navbar button:not(.float)");
//const buttons = document.querySelectorAll(".bottom-navbar button:not(.float)");
const effect = document.querySelector(".effect");
const container = document.querySelector(".container");
let y = 0;
let moveY = 0;
let open = false;

const vh = window.innerHeight * 0.01;
document.documentElement.style.setProperty("--vh", `${vh}px`);
setTimeout(function () {
  window.scrollTo(0, 1);
}, 0);

window.addEventListener("touchstart", (evt) => {
  const area = window.innerHeight - evt.touches[0].clientY;
  y = area;
  console.log(y);
});

window.addEventListener("touchend", (evt) => {
  y = 0;
  console.log(moveY);
  if (moveY > window.innerHeight / 4) {
    anime({
      targets: ".container",
      translateY: `-${window.innerHeight / 2}px`,
      duration: 600,
    });
    open = true;
  } else {
    anime({
      targets: ".container",
      translateY: `0px`,
      duration: 600,
      easing: "easeOutExpo",
    });
    open = false;
  }
});

window.addEventListener("touchmove", (evt) => {
  moveY = window.innerHeight - y - evt.touches[0].clientY;
  console.log(y);
  if (!open) {
    anime({
      targets: ".container",
      translateY: `${
        moveY <= window.innerHeight / 2
          ? moveY > 0
            ? -moveY
            : 0
          : -window.innerHeight / 2
      }px`,
      duration: 200,
    });
  } else if (open) {
    moveY = moveY + window.innerHeight / 2;
    anime({
      targets: ".container",
      translateY: `${
        moveY <= window.innerHeight / 2
          ? moveY > 0
            ? -moveY
            : 0
          : -window.innerHeight / 2
      }px`,
      duration: 200,
    });
  }
});

console.log("buttons: ",buttons)
buttons.forEach((item) => {
  console.log("item: ",item)
  item.addEventListener("click", (evt) => {
    const x = evt.target.offsetLeft;
    buttons.forEach((btn) => {
      btn.classList.remove("active");
    });
    evt.target.classList.add("active");
    anime({
      targets: ".effect",
      left: `${x}px`,
      duration: 600,
    });
  });
});

function handleClickPlus(evt) {
  anime({
    targets: ".container",
    translateY: `-${window.innerHeight / 2}px`,
    duration: 600,
  });
  open = true;
  y = window.innerHeight / 2;
  moveY = moveY + window.innerHeight / 2;
}

function handleClickPlusClose(evt) {
  anime({
    targets: ".container",
    translateY: `0px`,
    duration: 600,
    easing: "easeOutExpo",
  });
  open = false;
}

export default function Navbar({ children }) {
  return (
    <div>
      <div className="bottom-navbar">
        <div className="con-effect">
          <div className="effect"></div>
        </div>
        <button className="active">
          <i className="bx bx-home"></i>
        </button>
        <button>
          <i className="bx bx-skip-previous"></i>
        </button>
        <button onClick={handleClickPlus} className="float">
          <i className="bx bx-plus"></i>
        </button>
        <button>
          <i className="bx bx-skip-next"></i>
        </button>
        <button>
          <i className="bx bxs-save"></i>
        </button>
      </div>
      <div className="container">
        <img src="/pc.png" alt=""></img>
        <p>Your food instantly at your home, faster food tastier food</p>
        <button onClick={handleClickPlusClose}>Buy food</button>
        {children}
        <button onClick={handleClickPlusClose}>Close</button>
      </div>
    </div>
  );
}


src/components/subcomponents/editToolbar/EditToolbar.js
import { useEffect, useRef, useState } from "react";
import "./EditToolbar.css";

/*

function handleClickEdit(evt) {
  console.log("handleClickEdit", evt);
}
// https://reactgo.com/react-call-parent-component-method/
function handleClickDel(evt) {
  console.log("handleClickDel", evt);
}
function handleClickBack(evt) {
  console.log("handleClickBack", evt);
}
*/
export default function EditToolbar(props) {
  const { delCard, backCard, editCard } = props;
  useEffect(() => {
    /**
     * run after render
     */
  }, []);
  const [visible, setVisible] = useState(true);

  return visible ? (
    <div className="ItemPreviewPopOutButton">
      <button onClick={editCard} className="float">
        <i className="bx bx-pencil bx-xs"></i>
      </button>
      <button onClick={backCard} className="float">
        <i className="bx bx-arrow-back bx-xs"></i>
      </button>
      <button onClick={delCard} className="float">
        <i className="bx bx-trash bx-xs"></i>
      </button>
    </div>
  ) : (
    <div />
  );
}


src/scripts/cardsActions.js
import "./cardsActions.css";
/*
taphold | long press

https://interactjs.io/
https://jsfiddle.net/kelunik/pkjze6e6/42/
https://stackoverflow.com/questions/2625210/long-press-in-javascript
https://www.w3schools.com/jquerymobile/tryit.asp?filename=tryjqmob_events_taphold
https://stackoverflow.com/questions/41329964/click-and-hold-onto-a-selected-div-to-hide
https://www.npmjs.com/package/long-press-event


https://codepen.io/allurewebsolutions/pen/YqJyYY
https://codepen.io/borntofrappe/pen/jOEKERG

var node = document.getElementsByTagName("p")[0];
var longpress = false;
var presstimer = null;
var longtarget = null;

var cancel = function(e) {
    if (presstimer !== null) {
        clearTimeout(presstimer);
        presstimer = null;
    }
    
    this.classList.remove("longpress");
};

var click = function(e) {
    if (presstimer !== null) {
        clearTimeout(presstimer);
        presstimer = null;
    }
    
    this.classList.remove("longpress");
    
    if (longpress) {
        return false;
    }
    
    alert("press");
};

var start = function(e) {
    //console.log(e);
    
    if (e.type === "click" && e.button !== 0) {
        return;
    }
    
    longpress = false;
    
    this.classList.add("longpress");
    
    presstimer = setTimeout(function() {
        alert("long click");
        longpress = true;
    }, 1000);
    
    return false;
};

node.addEventListener("mousedown", start);
node.addEventListener("touchstart", start);
node.addEventListener("click", click);
node.addEventListener("mouseout", cancel);
node.addEventListener("touchend", cancel);
node.addEventListener("touchleave", cancel);
node.addEventListener("touchcancel", cancel);
*/
var pointerX = -1;
var pointerY = -1;
/*
document.addEventListener("DOMContentLoaded", function (event) {
  //we ready baby
  //https://www.codegrepper.com/code-examples/javascript/get+cursor+position+javascript


  onmousemove = function(e){
      //console.log("mouse location:", e.clientX, e.clientY);
      pointerX = event.clientX;
      pointerY = event.clientY;
    }
});
*/
//https://stackoverflow.com/questions/7790725/javascript-track-mouse-position
function tellPos(e) {
  //console.log("Position X : " + e.pageX + "<br />Position Y : " + e.pageY);
  //console.log("mouse location:", e.clientX, e.clientY);
  pointerX = e.clientX;
  pointerY = e.clientY;
}

document.addEventListener("mousemove", tellPos, false);

export default function onLongpress(el, handler) {
  //var node = document.getElementsByTagName("p")[0];
  const MAX_TIME_TO_BE_LONG = 1000; // espera máximo 1000ms (1 seg) al siguiente click/tap

  var longpress = false;
  var presstimer = null;
  var longtarget = null;

  // https://stackoverflow.com/questions/442404/retrieve-the-position-x-y-of-an-html-element
  // https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect
  //var bodyRect = document.body.getBoundingClientRect(), elemRect = el.getBoundingClientRect(), offset   = elemRect.top - bodyRect.top;

  //const elemRectY = el.getBoundingClientRect().y; // before elemRectTop
  //const elemRectX = el.getBoundingClientRect().x; // before elemRectLeft
  //alert('Element is ' + offset + ' vertical pixels from <body>');
  //console.log('Element is ' + offset + ' vertical pixels from <body>');

  function clearStyle(presstimer, element) {
    clearTimeout(presstimer);
    presstimer = null;
    element.classList.remove("longpress");
  }

  var cancel = function (e) {
    if (presstimer !== null) {
      clearTimeout(presstimer);
      presstimer = null;
    }

    this.classList.remove("longpress");
  };

  var click = function (e) {
    if (presstimer !== null) {
      clearTimeout(presstimer);
      presstimer = null;
    }

    this.classList.remove("longpress");

    if (longpress) {
      return false;
    }

    //alert("press");
    //console.log("press");
  };

  /**
   * todo 🚀
   * check if mouse move
   * check div height and width
   */

  var start = function (e) {
    console.log(
      "Element is moving " + el.getBoundingClientRect().top + " vertical pixels"
    );

    //console.log(e);

    if (e.type === "click" && e.button !== 0) {
      return;
    }

    longpress = false;

    this.classList.add("longpress");

    const elemRectY = el.getBoundingClientRect().y; // before elemRectTop
    const elemRectX = el.getBoundingClientRect().x; // before elemRectLeft
    const pointerRectY = pointerY;
    const pointerRectX = pointerX;

    presstimer = setTimeout(function () {
      //console.log("New tap");
      var elemRectYNow = el.getBoundingClientRect().y;
      var elemRectXNow = el.getBoundingClientRect().x;

      var pointerRectYNow = pointerY;
      var pointerRectXNow = pointerX;

      //console.log("elemRectY", elemRectY);
      //console.log("elemRectYNow", elemRectYNow);
      //console.log("elemRectX", elemRectX);
      //console.log("elemRectXNow", elemRectXNow);

      //console.log("pointerRectY", pointerRectY);
      //console.log("pointerRectYNow", pointerRectYNow);
      //console.log("pointerRectX", pointerRectX);
      //console.log("pointerRectXNow", pointerRectXNow);
      if (
        (elemRectY != elemRectYNow || elemRectX != elemRectXNow) &&
        (pointerRectY != pointerRectYNow || pointerRectX != pointerRectXNow)
      ) {
        //alert("is different");
        //console.log("is different");
        //var self = this;
        clearStyle(presstimer, el);
      } else {
        //alert("is not different");
        //alert("long click");
        console.log("is not different");
        el.classList.add("itemselected");
        handler(el); // ejecuta el handler
      }
      longpress = true;
    }, MAX_TIME_TO_BE_LONG);

    return false;
  };

  el.addEventListener("mousedown", start);
  el.addEventListener("touchstart", start);
  el.addEventListener("click", click);
  el.addEventListener("mouseout", cancel);
  el.addEventListener("touchend", cancel);
  el.addEventListener("touchleave", cancel);
  el.addEventListener("touchcancel", cancel);

  /*
    const MAX_TIME_TO_SECOND_TAP = 300 // espera máximo 300ms al siguiente click/tap
    let lastTapTimestamp = 0;
    el.addEventListener('touchstart', function() {
      lastTapTimestamp = new Date().getTime();
    });
    el.addEventListener('touchend', function(event) {
        var currentTime = new Date().getTime();
        var tapEllapsedTime = currentTime - lastTapTimestamp;
  
        // si se ha hecho el segundo tap/click entre los 300ms
        if (tapEllapsedTime < MAX_TIME_TO_SECOND_TAP && tapEllapsedTime > 0) {
            event.preventDefault();
            lastTapTimestamp = 0; // reinicia el counter
            handler(event); // ejecuta el handler
        }
    });
    */
}
